pkg/apis/v1/ec2nodeclass_conversion.go

```diff
@@ -20,16 +20,32 @@ import (
 	"strings"
 
 	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/api/resource"
 	"knative.dev/pkg/apis"
 
+	"github.com/aws/aws-sdk-go/service/ec2"
+
 	"github.com/aws/karpenter-provider-aws/pkg/apis/v1beta1"
 )
 
 func (in *EC2NodeClass) ConvertTo(ctx context.Context, to apis.Convertible) error {
 	v1beta1enc := to.(*v1beta1.EC2NodeClass)
 	v1beta1enc.ObjectMeta = in.ObjectMeta
+	v1beta1enc.Annotations = lo.OmitByKeys(v1beta1enc.Annotations, []string{AnnotationUbuntuCompatibilityKey})
+
+	if value, ok := in.Annotations[AnnotationUbuntuCompatibilityKey]; ok {
+		compatSpecifiers := strings.Split(value, ",")
+		// The only blockDeviceMappings present on the v1 EC2NodeClass are those that we injected during conversion.
+		// These should be dropped.
```

Comment:
There are two scenarios we here: updating via the v1 API or the v1beta1 API. If someone updates via the v1beta1 API, the annotation will be updated and the BDMs **won't** be removed when we convert back from v1 to v1beta1. If they make an update to the v1 resource, the annotation would not be updated, and they would be removed when converting back to v1beta1. The user should manually remove this annotation if they're applying the v1 resource. This will need to be called out in the docs.